src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -104,12 +104,11 @@ def __call__(
             latents = torch.randn(
                 latents_shape,
                 generator=generator,
-                device=self.device,
```

Comment:
this should not be removed here, this was added to get 1:1 results with stable diffusion cf https://github.com/huggingface/diffusers/pull/182#discussion_r945939774

Reply:
Ah, missed that, sorry. I saw that the behaviour in the rest of pipelines was different (`latents` are always moved to the device after generation, except here), so we might want to use a comment in this case.

Doing it like that doesn't work in `mps` (the `generator` is essentially ignored, I think), so for reproducibility, and to obtain the same results as when the CPU is used, we'll need to use a conditional in this case.